tests/bsim/bluetooth/mesh/overlay_psa.conf

```diff
@@ -5,3 +5,4 @@ CONFIG_BT_GATT_CACHING=n
 
 # Enable mbedTLS PSA as a crypto backend
 CONFIG_BT_MESH_USES_MBEDTLS_PSA=y
+CONFIG_PSA_CRYPTO_ENABLE_ALL=y
```

Comment:
I do not think I agree that BLE Mesh requires all crypto PSA features. There is list of features that Mesh uses, those are `selected` during `BT_MESH_USES_MBEDTLS_PSA` activation.

Reply:
There is one more thing. As a Bluetooth Mesh user, I expect that when I chose crypto backend, this is sufficient to build my application. This extra option seems like hidden knowledge that isn't obvious to people not known mbedtls-psa configuration system.

Reply:
> I do not think I agree that BLE Mesh requires all crypto PSA features. There is list of features that Mesh uses, those are `selected` during `BT_MESH_USES_MBEDTLS_PSA` activation.

So far the PSA configuration has been defined by MbedTLS's default configuration file, `psa/crypto_config.h`, which unconditionally does `#define PSA_WANT_[...] 1` for every symbol. In this regard this PR does not change anything. It introduces the configuration of `PSA_WANT` symbols for future work to build on top of this. Also note that when TF-M is in use, currently the actual crypto features available are defined by what TF-M profile is used.

However I am not certain how this fares when TF-M is not in use. It could be that with MbedTLS's adjustment of legacy symbols from PSA ones this would inflate the final code size for nothing.

> There is one more thing. As a Bluetooth Mesh user, I expect that when I chose crypto backend, this is sufficient to build my application. This extra option seems like hidden knowledge that isn't obvious to people not known mbedtls-psa configuration system.

Good point. Could we instead `select PSA_CRYPTO_ENABLE_ALL` @valeriosetti? (And you may also want to add it under `config BT_MESH_USES_TFM_PSA`, actually.) Or then, because of the first point above, select only needed PSA symbols. Or simply stick to the legacy ones and `MBEDTLS_USE_PSA_CRYPTO` if possible?

Reply:
As @alxelax suggested `BT_MESH_USES_MBEDTLS_PSA` selects required MbedTLS features, so I can try to replace legacy `MBEDTLS_` symbols with proper `PSA_WANT_` ones. This should fix both comments.
I'll give this a try

Reply:
@alxelax I reverted the previous change and I've added https://github.com/zephyrproject-rtos/zephyr/pull/72243/commits/513fb7d339a697eae7b7c1920123f0e909c927a9 which should match both your requirements. Wdyt?

Reply:
Yep, now it looks much better. Seems it missed some option(s) since devices cannot generate public key in bsim tests.

Reply:
Yes, I should have fixed both EC key generation issue and a missing  `PSA_WANT_ALG_ECB_NO_PADDING`. Hopefully bsim will be fine now.